### PR TITLE
Issue/1576 datapoint name

### PR DESCRIPTION
**Pull request by ichinaski**

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/app/web/DataProcessorRestServlet.java [hunk 1]
```diff
@@ -1520,18 +1524,32 @@ private void assembleDatapointName(Long surveyId, Long surveyedLocaleId) {
...
+            // Fetch all locales for this survey group
+            locales = slDao.listLocalesBySurveyGroupId(surveyGroupId);
         } else {
-            // get locales by createdSurveyId
-            locales = slDao.listLocalesByCreationSurvey(surveyId, Constants.ALL_RESULTS, null);
+            return;
+        }
+        
+        SurveyGroup sg = sgDao.getByKey(surveyGroupId);
+        if (sg == null) {
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/app/web/DataProcessorRestServlet.java]: I was thinking perhaps to make this cleaner you can already check for `surveyedlocaleid == null` , `surveyGroupId == null` and `sg == null` at the start of the method. Preconditions.

Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/app/web/DataProcessorRestServlet.java]: We can check that either `surveyedLocaleId` or `surveyGroupId` is not null, yes. Just note that we don't require both to be defined. Also `sg` is not known until later on, once the `SurveyGroup` object is fetched.

Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/app/web/DataProcessorRestServlet.java]: also if `sg.newLocaleSurveyId == null`

**Comment by muloem** [comment]
:+1: code reviewed!

**Pull request merged**